examples/seq2seq/seq2seq_trainer.py

```diff
@@ -41,12 +41,13 @@
 
 
 class Seq2SeqTrainer(Trainer):
-    def __init__(self, config, data_args, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
```

Comment:
@patil-suraj @sshleifer - I think it would be better to align the init of `Seq2SeqTrainer` 100% with `Trainer`.
Is there a reason why we would insert config instead of using the model's config? 

Also I don't really think the variable `data_args` is necessary. Both `max_length` and `num_beams` can be defined in the config and don't have to be "force" passed to the `generate()`  method.

Reply:
`model.config` breaks under `DistributedDataParallel`, so we decided to pass it explicitly. See #7461 and #7460.

if default `num_beams`  and `max_length` is too high it'll slow down evaluation, so we allow the use to control it during training. And not overriding `config` since defaults will be needed for inference after training.

Reply:
Okok I see! I'm a bit confused why `Trainer` does not break with `DistributedDataParallel` when only using `model.config....` , but `Seq2SeqTrainer` does? Do you guys know why?

Reply:
> `eval_beams`/`eval_max_gen_length` reasoning:
> @patil-suraj said exactly this LOL, but in my words:
> users are not good at modifying configs locally. We want to have a way to run `num_beams=2` during the generation step, but then end up with a trained model with the default # beams. In general, we try not to manipulate config attributes that would only be desired during training.

I mean modifying the configs locally is as simple as `config.num_beams = 4` and I would think one wants to evaluate a model during training with exactly the beam size and max_length that is stored in the config (I mean changing the beam size and max_length does not simple reduce time, but also changes the output...) But I guess I can see the use case where the people want to tweak `max_length` and `num_beams` without changing the config. Would it be fine to make `data_args` optional and call them `generation_args` that will just be passed as **generation_args to the generate function?

Reply:
ups that was supposed to land further below not here. @sshleifer for reference.

Reply:
Whether they are called `data_args` -> `generation_kwargs` seems like a good change, but the CLI naming has a purpose:
It wouldn't have been obvious to me that passing `--min_length 32` would affect generation, rather than truncating source docs. That's why the `eval_` prefix was added.